lib/services/storageManagement2/lib/operations/storageAccounts.js

```diff
@@ -39,10 +41,42 @@ function StorageAccounts(client) {
  * @param {object} [options.customHeaders] Headers that will be added to the
  * request
  *
- * @param {function} callback
+ * @returns {Promise} A promise is returned
+ * @resolve {HttpOperationResponse<CheckNameAvailabilityResult>} - The deserialized result object.
+ * @reject {Error} - The error object.
+ */
+StorageAccounts.prototype.checkNameAvailabilityWithHttpOperationResponse = function (name, options) {
+  let self = this;
+  let client = self.client;
+  return new Promise((resolve, reject) => {
+    self._checkNameAvailability(name, options, (err, result, request, response) => {
+      if (err) {
+        return reject(err);
+      }
+      let httpOperationResponse = new HttpOperationResponse(request, response);
+      httpOperationResponse.body = result;
+      return resolve(httpOperationResponse);
```

Comment:
That makes sense because the callback doesn't end the current function.  So yeah, it might be good evening just based on the semantics of your expectation for execution to end to keep the return statements.